### PR TITLE
[10802] Collection of data and notification to listeners PDP_PACKETS

### PR DESCRIPTION
**Pull request by MiguelBarro**
There is a little piggyback preventing the dds layer statistics endpoints from been instrumented from the rtps layer.

### CODE REVIEW

File: src/cpp/statistics/types/typesPubSubTypes.cxx [hunk 1]
```diff
@@ -1279,9 +1279,9 @@ namespace eprosima {
             Entity2LocatorTrafficPubSubType::Entity2LocatorTrafficPubSubType()
             {
                 setName("eprosima::fastdds::statistics::Entity2LocatorTraffic");
-                auto type_size = Entity2LocatorTraffic::getMaxCdrSerializedSize();
-                type_size += eprosima::fastcdr::Cdr::alignment(type_size, 4); /* possible submessage alignment */
-                m_typeSize = static_cast<uint32_t>(type_size) + 4 /*encapsulation*/;
+                auto type_size = static_cast<uint32_t>(Entity2LocatorTraffic::getMaxCdrSerializedSize());
+                type_size += static_cast<uint32_t>(eprosima::fastcdr::Cdr::alignment(type_size, 4)); /* possible submessage alignment */
+                m_typeSize = type_size + 4 /*encapsulation*/;
```
Comment (MiguelCompany) [inline src/cpp/statistics/types/typesPubSubTypes.cxx:1284]: What was wrong in the original code? Method `eprosima::fastcdr::Cdr::alignment()` takes and returns a `size_t` ...

File: src/cpp/rtps/participant/RTPSParticipantImpl.cpp [hunk 2]
```diff
@@ -2012,10 +2017,13 @@ bool RTPSParticipantImpl::register_in_reader(
...
         {
-            res &= reader->add_statistics_listener(listener);
+            if (!fastdds::statistics::is_statistics_builtin(reader->getGuid().entityId))
+            {
+                res &= reader->add_statistics_listener(listener);
+            }
         }
     }
-    else
+    else if (!fastdds::statistics::is_statistics_builtin(reader_guid.entityId))
```
Comment (MiguelCompany) [inline src/cpp/rtps/participant/RTPSParticipantImpl.cpp:2026]: Statistics builtin endpoints will never be readers.

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/participant/RTPSParticipantImpl.h [hunk 3]
```diff
@@ -293,6 +293,9 @@ class RTPSParticipantImpl
                 destination_locators_begin,
                 destination_locators_end,
                 msg->length);
+
+            // checkout if sender is a pdp endpoint
+            on_pdp_packet(sender_guid);
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply src/cpp/rtps/participant/RTPSParticipantImpl.h:298]: I agree

File: src/cpp/statistics/types/typesPubSubTypes.cxx (hunk 1 above)
Comment (MiguelBarro) [reply src/cpp/statistics/types/typesPubSubTypes.cxx:1284]: Nothing, we both corrected the same warning in different ways. I will fetch yours.

File: src/cpp/rtps/participant/RTPSParticipantImpl.cpp (hunk 2 above)
Comment (MiguelBarro) [reply src/cpp/rtps/participant/RTPSParticipantImpl.cpp:2026]: just in case 😂

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Comment by MiguelCompany** [comment]
@richiprosima Please test this 🤞

**Pull request merged**